src/main/assets/js/courts-table-search.ts

```diff
@@ -0,0 +1,220 @@
+import $ from 'jquery';
+import {orderToggleState} from '../../enums/searchToggleState';
+
+export class CourtsTableSearch {
+
+  private static toggleClosedCourtsDisplay = 'toggleClosedCourtsDisplay';
+  private static numberOfCourts = '#numberOfCourts';
+  private static searchCourtsFilterId = '#searchCourts';
+  private static courtsHiddenId = 'courtsHidden';
+  private static courtsNameAscToggleId = '#courtsNameAscToggle';
+  private static courtsUpdatedAscToggleId = '#courtsUpdatedAscToggle';
+  private static courtsResultsSection = '#courtResults > tbody';
+  private static courtsTableHeaderAsc = 'courts-table-header-asc';
+  private static courtsTableHeaderDesc = 'courts-table-header-desc';
+  private static courtsTableHeaderInactive = 'courts-table-header-inactive';
+  private static courtsFrontendUrl = '#courtsFrontendUrl';
+  private static tableCourtsNameId = '#tableCourtsName';
+  private static tableCourtsUpdatedId = '#tableCourtsUpdated';
+
+  /**
+   *
+   * Sets up the table view for the main page, on load the default configuration
+   * is used, otherwise it is updated on each search toggle or courts search
+   *
+   * @param filterName Either 'name' or 'date' depending on the type of toggle
+   * @param defaultValue set to false by default on initial load
+   */
+  static setUpTable(filterName: string, defaultValue = false): void {
```

Comment:
If filterName has to be a string, I think you can reduce the type to ` "name" | "date" `